File: compute-node/src/__tests__/translator/utility.test.ts
import { describe, expect, it } from "vitest";
import { translate } from "../../translator";
import type { ResultRow } from "../../translator/types";

describe("utility translator", () => {
  it("PING returns PONG", () => {
    const translation = translate(["PING"]);
    expect(translation.statements).toHaveLength(0);
    const results: ResultRow[][] = [];
    expect(translation.mapResult(results)).toBe("PONG");
  });
});

File: compute-go/translator/utility_test.go
package translator

import "testing"

func TestTranslatePing(t *testing.T) {
	translation, err := Translate([]string{"PING"})
	if err != nil {
		t.Fatalf("Translate error: %v", err)
	}
	if len(translation.Statements) != 0 {
		t.Fatalf("expected 0 statements, got %d", len(translation.Statements))
	}
	value, err := translation.MapResult(nil)
	if err != nil {
		t.Fatalf("MapResult error: %v", err)
	}
	if value != "PONG" {
		t.Fatalf("expected PONG, got %v", value)
	}
}
